samples/extensions/fragment_shader_barycentric/fragment_shader_barycentric.cpp

```diff
@@ -0,0 +1,404 @@
+
+	// Use the pNext to point to the rendering create struct
+	VkGraphicsPipelineCreateInfo graphics_create{VK_STRUCTURE_TYPE_GRAPHICS_PIPELINE_CREATE_INFO};
+	graphics_create.pNext               = VK_NULL_HANDLE;
+	graphics_create.renderPass          = render_pass;
+	graphics_create.pInputAssemblyState = &input_assembly_state;
+	graphics_create.pRasterizationState = &rasterization_state;
+	graphics_create.pColorBlendState    = &color_blend_state;
+	graphics_create.pMultisampleState   = &multisample_state;
+	graphics_create.pViewportState      = &viewport_state;
+	graphics_create.pDepthStencilState  = &depth_stencil_state;
+	graphics_create.pDynamicState       = &dynamic_state;
+	graphics_create.pVertexInputState   = &vertex_input_state;
+	graphics_create.stageCount          = static_cast<uint32_t>(shader_stages.size());
+	graphics_create.pStages             = shader_stages.data();
+	graphics_create.layout              = pipeline_layout;
+
+	// Skybox pipeline (background cube)
+	vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipelines.skybox);
+
+	// Object pipeline
+	shader_stages[0] = load_shader("fragment_shader_barycentric/object.vert", VK_SHADER_STAGE_VERTEX_BIT);
+	shader_stages[1] = load_shader("fragment_shader_barycentric/object.frag", VK_SHADER_STAGE_FRAGMENT_BIT);
+
+	// Flip cull mode
+	rasterization_state.cullMode = VK_CULL_MODE_FRONT_BIT;
+	vkCreateGraphicsPipelines(get_device().get_handle(), pipeline_cache, 1, &graphics_create, VK_NULL_HANDLE, &pipelines.object);
+}
+
+/**
+ *  @fn void FragmentShaderBarycentric::draw()
+ *  @brief Preparing frame and submitting it to the present queue
+ */
+void FragmentShaderBarycentric::draw()
+{
+	ApiVulkanSample::prepare_frame();
+	submit_info.commandBufferCount = 1;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+/**
+ * 	@fn void FragmentShaderBarycentric::build_command_buffers()
+ * 	@brief Creating command buffers and drawing background and model on window
+ */
+void FragmentShaderBarycentric::build_command_buffers()
+{
+	std::array<VkClearValue, 2> clear_values{};
+	clear_values[0].color        = {{0.0f, 0.0f, 0.0f, 0.0f}};
+	clear_values[1].depthStencil = {0.0f, 0};
+
+	int i = -1;
+	for (auto &draw_cmd_buffer : draw_cmd_buffers)
+	{
+		i++;
+		auto command_begin = vkb::initializers::command_buffer_begin_info();
+		VK_CHECK(vkBeginCommandBuffer(draw_cmd_buffer, &command_begin));
+
+		VkRenderPassBeginInfo render_pass_begin_info    = vkb::initializers::render_pass_begin_info();
+		render_pass_begin_info.renderPass               = render_pass;
+		render_pass_begin_info.framebuffer              = framebuffers[i];
+		render_pass_begin_info.renderArea.extent.width  = width;
+		render_pass_begin_info.renderArea.extent.height = height;
+		render_pass_begin_info.clearValueCount          = 2;
```

Comment:
That is definitely a better option than hardcoded value. This will help avoid mistakes like the one I made before (where I inserted 3 instead of 2). Thanks for your comments!